pytorch_lightning/core/optimizer.py

```diff
@@ -207,7 +207,7 @@ def closure_dis():
             profiler_name = "closure_{self._optimizer_idx}"
             closure = do_nothing_closure
         else:
-            if not isinstance(closure, types.FunctionType):
+            if not isinstance(closure, Callable):
```

Comment:
those checks don't work for all cases. Better replace it with `callable(closure)`
```suggestion
            if not callable(closure):
```